Stripe/STPURLCallbackHandler.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  STPURLCallbackHandler.swift
+//  Stripe
+//
+//  Created by Brian Dorfman on 10/6/16.
+//  Copyright © 2016 Stripe, Inc. All rights reserved.
+//
+
+import Foundation
+
+@objc protocol STPURLCallbackListener: NSObjectProtocol {
+  func handleURLCallback(_ url: URL) -> Bool
+}
+
+class STPURLCallbackHandler: NSObject {
+  static var sharedHandler: STPURLCallbackHandler = STPURLCallbackHandler()
+
+  @objc class func shared() -> STPURLCallbackHandler {
+    return sharedHandler
+  }
+
+  @objc @discardableResult func handleURLCallback(_ url: URL) -> Bool {
+    let components = NSURLComponents(
+      url: url,
+      resolvingAgainstBaseURL: false)
+
+    var resultsOrred = false
+
+    for callback in callbacks {
+      if let components = components {
+        if callback.urlComponents?.stp_matchesURLComponents(components) ?? false {
+          resultsOrred = resultsOrred || (callback.listener?.handleURLCallback(url) ?? false)
```

Comment:
is this gonna short circuit and not call additional handlers?

Reply:
I don't think so, but I cleaned it up anyway